### PR TITLE
libceed - update libCEED headers and version

### PR DESCRIPTION
**Pull request by jeremylt**
libCEED is releasing v0.8 later today! I think that I handled all of the appropriate changes.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2142](https://github.com/mfem/mfem/pull/2142) | @jeremylt | @tzanio | @barker29 + @YohannDudouit | 04/01/21 | 04/01/21 | 04/07/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
Please remember to label with`ready-for-review` when ready

**Review by barker29 — APPROVED** [review]
Looks good, thanks Jeremy.

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by YohannDudouit** [comment]
I think you missed `<ceed-hash.h>` -> `<ceed/hash.h>`:
```
./../libCEED/include/ceed-hash.h:3:2: warning: <ceed-hash.h> is deprecated and will be removed in libCEED v0.9; use <ceed/hash.h> [-W#warnings]
#warning <ceed-hash.h> is deprecated and will be removed in libCEED v0.9; use <ceed/hash.h>
```

File: fem/ceed/util.hpp [hunk 1]
```diff
@@ -15,8 +15,8 @@
 #include "../../config/config.hpp"
 #ifdef MFEM_USE_CEED
 #include <ceed.h>
-#include <ceed-hash.h>
-#include <ceed-backend.h>  // for CeedOperatorField
+#include <ceed/hash.h>
```
Comment (jeremylt) [inline fem/ceed/util.hpp:18]: I only found `ceed-hash.h` here. Is there a second location?

Comment (YohannDudouit) [reply fem/ceed/util.hpp:18]: My bad............

**Review by YohannDudouit — APPROVED** [review]
Looks good to me, thank you @jeremylt for doing this.

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by tzanio** [comment]
LGTM

**Comment by tzanio** [comment]
Re-merged in `next` for testing ...

**Pull request merged**